tests/integration/integration_test.go

```diff
@@ -16,49 +17,30 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-// // small set of actions to trigger a magic write event
-// func checkMagicwrite(t *testing.T, gotOutput *[]trace.Event) {
-// 	// create a temp dir for testing
-// 	d, err := ioutil.TempDir("", "Test_MagicWrite-dir-*")
-// 	require.NoError(t, err)
-
-// 	// cp a file to trigger
-// 	f, err := os.CreateTemp(d, "Test_MagicWrite-file-*")
-// 	require.NoError(t, err)
-// 	defer func() {
-// 		os.Remove(d)
-// 	}()
-
-// 	f.WriteString(`foo.bar.baz`)
-// 	f.Close()
-
-// 	cpCmd := exec.Command("cp", f.Name(), filepath.Join(d+filepath.Base(f.Name())+"-new"))
-// 	fmt.Println("executing: ", cpCmd.String())
-// 	cpCmd.Stdout = os.Stdout
-// 	assert.NoError(t, cpCmd.Run())
-
-// 	waitForTraceeOutput(t, gotOutput, time.Now(), true)
-
-// 	// check tracee output
-// 	expect := []byte{102, 111, 111, 46, 98, 97, 114, 46, 98, 97, 122}
-// 	fail := true
-// 	for _, evt := range *gotOutput {
-// 		arg := events.GetArg(&evt, "bytes")
-// 		argVal, ok := arg.Value.([]byte)
-// 		require.Equal(t, true, ok)
-// 		ok = assert.ElementsMatch(t, argVal, expect)
-// 		if ok {
-// 			fail = false
-// 		}
-// 	}
-// 	if fail {
-// 		t.Fail()
-// 	}
-// }
+// small set of actions to trigger a magic write event
+func checkMagicwrite(t *testing.T, gotOutput *[]trace.Event) {
+
+	_, err := forkAndExecFunction(doMagicWrite)
+	require.NoError(t, err)
+
+	waitForTraceeOutput(t, gotOutput, time.Now(), true)
+
+	// check tracee output
+	fail := true
+	for _, evt := range *gotOutput {
+		ok := assert.ElementsMatch(t, []byte(evt.EventName), []byte("magic_write"))
```

Comment:
Why are you comparing this way and not just
```suggestion
		ok := assert.Equal(t, evt.Eventname, "magic_write")
```
Also am I reading it correct that if you find magic_write then the test fails? Shouldn't it be the opposite? If not we can also break on the `if ok` check.

Reply:
It is the opposite because fail is set to false. It is a bit backwards but it makes sense here to fail unless all criteria marks it as success.

Reply:
Good point, I cleaned it up